syntax = "proto3";

package tasks;

message Task {
  int32 id = 1;
  string title = 2;
  string description = 3;
  string author_id = 4;
  bool completed = 5;
}

message CreateTaskRequest {
  string title = 1;
  string description = 2;
  string author_id = 3;
}

message UpdateTaskRequest {
  Task task = 1;
}

message DeleteTaskRequest {
  int32 id = 1;
}

message GetTaskRequest {
  int32 id = 1;
}

message ListTasksRequest {
  int32 page = 1;
  int32 per_page = 2;
}

message ListTasksResponse {
  repeated Task tasks = 1;
}

service TaskService {
  rpc CreateTask(CreateTaskRequest) returns (Task);
  rpc UpdateTask(UpdateTaskRequest) returns (Task);
  rpc DeleteTask(DeleteTaskRequest) returns (Empty);
  rpc GetTask(GetTaskRequest) returns (Task);
  rpc ListTasks(ListTasksRequest) returns (ListTasksResponse);
}

message Empty {}